### ISSUE TITLE
Removing "(required)" with options that have a default value specified

### ISSUE THREAD

**Issue by sanks64** [devnote-dev/cling#9]
Example:

```
Options:
	-i, --input-path       Specify the path to the input file (required)
	-o, --output-path      Specify the path to the output directory (required) (default: out)
```

I think having "(required)" makes sense in the first case, but not the second one.

Alternatively, it would make sense if passing `-o` without a value would cause an error with the same message as when `-i` is missing:

```
Unhandled exception: Missing required argument for option 'i' (Cling::ExecutionError)
```

**Comment by devnote-dev** [comment]
I think the bigger issue here is whether required options should be allowed to have default values. With the current implementation, the `Executor` will just set the value of a required option to the default, so it isn't _really_ a required option. At the moment, I'm inclined to disallow required options with defaults for the above reason, but I'm open to other views.

> Alternatively, it would make sense if passing `-o` without a value would cause an error with the same message as when `-i` is missing:

I agree, and this should probably be the case for optional options with defaults.

**Comment by devnote-dev** [comment]
Implemented in 865bb48aa85c81e8b26ad3c35d7d60ff07228f23 and ad15353213db78fd8e0ab29109c7a9c5b48dffb5. 👍
